builder/slide_and_fold_storage.cc

```diff
@@ -667,7 +667,7 @@ class slide_and_fold : public stmt_mutator {
       for (dim_fold_info& j : *i) {
         if (!depends_on(j.factor, l.worker_count).any()) continue;
 
-        if (l.data_parallel && max_workers == loop::parallel) {
+        if (l.data_parallel && max_workers != loop::serial) {
```

Comment:
Not sure about this. `max_workers < loop::parallel` is a loop, but it's not data parallel, the loop iterations have dependencies.

Reply:
We could define a loop with two workers instead of loop::parallel and that's still going to be a parallel loop, but its data_parallel is defined separately in the slide_and_fold_buffer, so it won't be safe in that case.